### PR TITLE
Basic gameplay mechanics & eventing system 

### PR DESCRIPTION
**Pull request by SrRyan**
This PR introduces a few new things 

- Some small reworking on the Interfaces to be easier to work with
- Clock Object & Interface to track in-game time 
- Business object with getters & setters
- HUD Scene 
- Eventing system 

The eventing system is barebones but allows us to talk cross-scene to enable gameplay behaviors. It is also how I imagine we will invoke random events and have the UI react.

Level one has some basic buttons to show how the communication works with cross-scene elements. I'm probably going to follow up with a bit more structure. 

I also removed references of the month from some of the data values as I found it already cumbersome to work with the data when in reality we can just do that ad-hoc if needed. It's better to give a real-time state I found.

demo1

### CODE REVIEW

File: src/events/eventCenter.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+export enum UIEvents {
+    UI_UPDATE_CASH = 'ui.update.cash',
+    UI_UPDATE_COSTS = 'ui.update.costs',
+}
+
+export enum GameplayBusinessEvents {
+    BUSINESS_UPDATE_CASH = 'gameplay.business.update.cash',
+    BUSINESS_UPDATE_COSTS = 'gameplay.business.update.costs',
+    BUSINESS_ADD_CUSTOMER = 'gameplay.business.add.customer',
+    BUSINESS_REMOVE_CUSTOMER = 'gameplay.business.remove.customer',
```
Comment (bethanyj28) [inline src/events/eventCenter.ts:12]: Maybe this is temporary - but I was under the impression that the game should manage the addition/removal of customers and the player wouldn't have control of this. However, the player should be able to manage employees. Would love your thoughts!

**Review by bethanyj28 — COMMENTED** [review]
Question about a field! Other than that I think this is awesome 🚀

File: src/events/eventCenter.ts (hunk 1 above)
Comment (bethanyj28) [reply src/events/eventCenter.ts:12]: Clarified in discord!

**Review by bethanyj28 — APPROVED** [review]

File: src/events/eventCenter.ts (hunk 1 above)
Comment (SrRyan) [reply src/events/eventCenter.ts:12]: As discussed in discord, this for now is just to show that we can show some sort of eventing and reaction between the different loops!

**Pull request merged**